sass/partials/components/_button.scss

```diff
@@ -3,15 +3,62 @@
 
 .button {
   @include button();
+  margin: 0 0 0.5em;
 
   & + & {
-    margin-left: 1em; // LTR
+    margin-left: 0.5em; // LTR
 
     @if $support-for-rtl {
       [dir='rtl'] & {
         margin-left: 0;
-        margin-right: 1em;
+        margin-right: 0.5em;
       }
     }
   }
 }
+
+.button--danger {
```

Comment:
Do we need all of these extra button styles?  Are they just examples or are any of these classes actually output in Drupal?